frontend/awx/common/adapters/awxErrorAdapter.tsx

```
@@ -42,3 +42,29 @@ export const awxErrorAdapter = (error: unknown): ErrorOutput => {
   }
   return { genericErrors, fieldErrors };
 };
+
+/**
+ * Parses the error message and returns a formatted message along with parsed errors.
+ * @param {Error} error - The error object to parse.
+ * @param {string} [unknownErrorMessage] - Optional. The translated string for unknown errors.
+ * @returns {{ message: string; parsedErrors: (GenericErrorDetail | FieldErrorDetail)[] }} An object containing the formatted error message and parsed errors.
+ */
+export function useAwxErrorMessageParser() {
+  return (
+    error: Error,
+    unknownErrorMessage?: string
+  ): { message: string; parsedErrors: (GenericErrorDetail | FieldErrorDetail)[] } => {
+    const { genericErrors, fieldErrors } = awxErrorAdapter(error);
+    const parsedErrors = [
+      ...genericErrors,
+      ...fieldErrors.filter((e) => e.message).map(({ message }) => ({ message })),
```

@jamestalton here we could just return the errors without being parsed.